validator/tests/test_scheduler/yaml_scheduler_tester.py

```diff
@@ -191,8 +192,14 @@ def run_scheduler(self,
                     txns_to_process.append(txn_info)
                 else:
                     stop = True
-            t_info = txns_to_process.pop()
+
+            if txns_executed_fifo:
```

Comment:
If you are going to pop from either end, should you switch `txns_to_process` to a `deque` instead?  I don't have a handle on how large this list might be, but popping from the front is not efficient...and you have chosen that as the default.